language/move-prover/bytecode/src/global_invariant_instrumentation_v2.rs

```diff
@@ -365,8 +362,10 @@ impl<'a> Instrumenter<'a> {
         entrypoint_invariants: &BTreeSet<GlobalId>,
     ) {
         // translate all the invariants. Some were already translated at the entrypoint, but
-        // that's ok because they are global invariants that don't have "old", so redundant
-        // state tags are not going to be a problem
+        // that's ok because entrypoint invariants are global invariants that don't have "old",
+        // so redundant state tags are not going to be a problem.
+        // TODO: can't we just translate the global_assumes,
+        // which are separated in a little while, anyway?
```

Comment:
Yes, we should just translate the assumes, so I guess the following code should do the job:

```rust
let (global_assumes, _update_invs) = self.separate_update_invariants(&modified_invs);
let modified_assumes: BTreeSet<_> = global_assumes
            .difference(entrypoint_invariants)
            .cloned()
            .collect();

let xlated_invs = SpecTranslator::translate_invariants_by_id(
  self.options.auto_trace_level.invariants(),
  &mut self.builder,
  &modified_assumes,
);

for (loc, mid, cond) in xlated_invs {
  self.emit_invariant(loc, cond, PropKind::Assume);
}

```

Reply:
The code is trickier than it looks.  It will need to be rewritten to fix some other things, too, so I'd like to defer cleanup and do that separately, when priorities dictate.